### PR TITLE
[aws-lambda] add types for CloudWatch alarm events

### PR DESCRIPTION
**Pull request by gfemec**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://aws.amazon.com/about-aws/whats-new/2023/12/amazon-cloudwatch-alarms-lambda-change-action/ https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarms-and-actions
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

Note: I didn't find a complete schema for these events in the AWS docs so these types are based on the examples included in the AWS docs and the results of me testing various alarms.

### CODE REVIEW

File: types/aws-lambda/trigger/cloudwatch-alarm.d.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+export interface CloudWatchAlarmEvent {
+    source: string;
+    alarmArn: string;
+    accountId: string;
+    time: string;
+    region: string;
+    alarmData: CloudWatchAlarmData;
+}
+
+export type CloudWatchAlarmHandler = Handler<CloudWatchAlarmEvent, void>;
```
Comment (simonbuchan) [inline types/aws-lambda/trigger/cloudwatch-alarm.d.ts:69]: Nit-iest of nits, but we try to have the Handler type at the top, followed by the Event (and Result if present).

**Review by simonbuchan — APPROVED** [review]
Great work, thanks!

File: types/aws-lambda/trigger/cloudwatch-alarm.d.ts (hunk 1 above)
Comment (gfemec) [reply types/aws-lambda/trigger/cloudwatch-alarm.d.ts]: No worries, happy to follow the convention. I've pushed an update.

**Review by simonbuchan — APPROVED** [review]

**Review by sandersn — APPROVED** [review]

**Comment by gfemec** [comment]
Ready to merge

**Pull request merged**